packages/api/src/add-build.js

```diff
@@ -125,15 +126,21 @@ async function addBuild (testCases, buildInfo, client, collectionName = 'reposit
     // update information for flakybuild and flakyrepo;
     flakyBuild = (buildtestCaseAnalytics.computeIsFlaky() && !testCase.successful) ? 1 + flakyBuild : flakyBuild;
     flakyRepo = (updateObj.flaky) ? 1 + flakyRepo : flakyRepo;
+    alltests.push({
+      name: testCase.name,
+      passed: testCase.successful,
+      flaky: updateObj.flaky,
```

Comment:
If we change our algorithm for flakiness over time, is it a problem that we'd have historic tests that were flagged as flaky using the old approach?

Reply:
Yes, when looking at a particular historical build, it's tests will be marked according to their status at the time of the build being posted. This means that even when you look at a historical build, it may be listed as flaky even if it has since been fixed. Note that metrics for a repository and for an individual test case will be updated.

The potential solutions are
1. Keeping this and when viewing a build see (Flaky as of this build, percent passing up to this build)
2. Not display this information for a particular build since it could be potentially confusing.
3. When viewing a particular build, see the most up to date flaky status and percent passing status. This overall seems better but has potential to be confusing. 

To accomplish (3.), it will involve no longer storing a list of all tests for a particular build. Then, when getting a particular build, iterating through the list of all test cases and querying .doc(<test name>).collection('runs').doc(<buildId>)

Reply:
we discussed this in chat:

I'm comfortable with `1.`, since builds represent a historical snapshot of the state of the tests at the time of the build. Whereas, if I'm looking at tests in aggregate, it should represent the current state of tests.